lib/src/cli.rs

```diff
@@ -903,10 +903,14 @@ impl Opt {
             let first: OsString = first.into();
             let argv0 = callname_from_argv0(&first);
             tracing::debug!("argv0={argv0:?}");
-            if argv0 == InternalsOpts::GENERATOR_BIN {
-                let base_args = ["bootc", "internals", "systemd-generator"]
-                    .into_iter()
-                    .map(OsString::from);
+            let mapped = match argv0 {
+                InternalsOpts::GENERATOR_BIN => {
+                    Some(["bootc", "internals", "systemd-generator"].as_slice())
```

Comment:
What is `.as_slice()` doing here?  I'd have assumed that it already is a slice...  Copying?  Could you rather do that with a ref or something?

Reply:
This coerces the arrays to slices so both values from the match have the same type.